### PR TITLE
Issue 16728

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java [hunk 1]
```diff
@@ -4188,6 +4188,85 @@ public void testUniqueTextFieldContentletsWithDiffLanguages()
         }
     }
 
+    @Test(expected = DotContentletValidationException.class)
+    public void testUniqueTextFieldContentletsWithSpecialChars()
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java]: good one for dataproviders

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 2]
```diff
@@ -5180,19 +5180,16 @@ else if(field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())){
 
                     buffy.append(" +(live:true working:true)");
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: I don't know if we handle searching both live and working in the same query - we might have to do two queries to be sure.

Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: done

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 3]
```diff
@@ -5172,25 +5172,21 @@ else if(field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())){
...
 
                     buffy.append(" +languageId:" + contentlet.getLanguageId());
 
                     buffy.append(" +" + contentlet.getContentType().variable() + "." + field
-                            .getVelocityVarName() + ESUtils.SHA_256 + ":");
-                    buffy.append(ESUtils.sha256(contentlet.getContentType().variable()
-                                    + "." + field.getVelocityVarName(),
-                            getFieldValue(contentlet, new LegacyFieldTransformer(field).from()),
-                            contentlet.getLanguageId()));
+                            .getVelocityVarName() + ":");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java]: Use the constant from StringPool

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**